Sources/AppIcon/main.swift

```diff
@@ -1,35 +1,56 @@
 import AppIconCore
-import Commander
-
-let main = command(
-    Argument<String>("base image (1024x1024.png)", description: "The path of the base image"),
-    Option("icon-name", default: "AppIcon", description: "The name of the generated image"),
-    Option("output-path", default: "AppIcon", description: "The path of the generated appiconset"),
-    Flag("ipad", description: "Whether or not to generate iPad icons"),
-    Flag("mac", description: "Whether or not to generate Mac icons"),
-    Flag("watch", description: "Whether or not to generate Apple Watch icons")
-) { base, iconName, path, ipad, mac, watch in
-    guard base.hasSuffix(".png") else {
-        throw ArgumentError.missingValue(argument: "base image (1024x1024.png)")
-    }
+import ArgumentParser
 
-    let outputExpansion = ".appiconset"
-    let outputPath = path.hasSuffix(outputExpansion) ? path : "\(path)\(outputExpansion)"
-    let platforms = Platform.platforms(ipad: ipad, mac: mac, watch: watch)
+struct AppIcon: ParsableCommand {
+    static let configuration = CommandConfiguration(
+        commandName: "appicon",
+        abstract: "AppIcon generates *.appiconset contains each resolution image for iOS",
+        version: "1.0.5"
+    )
 
-    do {
-        try ImageExtractor.extract(input: (base, platforms), output: (iconName, outputPath))
-    } catch {
-        print("Image Extraction Error has occured 😱")
-    }
+    @Argument(help: "The path to the base image (1024x1024.png)")
+    var image: String
 
-    do {
-        try JSONExtractor.extract(input: platforms, output: (iconName, outputPath))
-    } catch {
-        print("JSON Extraction Error has occured 😱")
-    }
+    @Option(help: "The name of the generated image")
+    var iconName = "AppIcon"
+
+    @Option(help: "The path of the generated appiconset")
+    var outputPath = "AppIcon"
+
+    @Flag(help: "Whether or not to generate iPad icons")
+    var ipad = false
+
+    @Flag(help: "Whether or not to generate Mac icons")
+    var mac = false
 
-    print("\(outputPath) has been generated 🎉")
+    @Flag(help: "Whether or not to generate Apple Watch icons")
+    var watch = false
+
+    func run() throws {
+        guard image.hasSuffix(".png") else {
+            throw ValidationError("image path should have .png extension")
+        }
+
+        let outputExpansion = ".appiconset"
+        let outputPath = self.outputPath.hasSuffix(outputExpansion) ? self.outputPath : "\(self.outputPath)\(outputExpansion)"
+        let platforms = Platform.platforms(ipad: ipad, mac: mac, watch: watch)
+
+        do {
+            try ImageExtractor.extract(input: (image, platforms), output: (iconName, outputPath))
+        } catch {
+            print("Image Extraction Error has occured 😱")
+            throw ExitCode(1)
```

Comment:
I also think in that and L48 cases program flow should be aborted to prevent wrong success info printing